docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md

```diff
@@ -0,0 +1,126 @@
+# Migration guide from 3.0.x to 3.1.x
+
+**Make sure your server is not running until the end of the migration**
+
+## Summary
+
+1. [Upgrading your dependencies](#_1-upgrading-your-dependencies)
+2. [Define the admin JWT Token](#_2-define-the-admin-jwt-token)
+3. [Migrate your custom admin panel plugins](#_3-migrate-your-custom-admin-panel-plugins)
+4. [Rebuild the admin panel](#_4-rebuild-the-admin-panel)
+
+## 1. Upgrading your dependencies
+
+First, run the following command to get the last version of `3.1.x`:
+
+```bash
+npm info strapi@3.1.x version
+```
+
+Then, update your `package.json` with the highest version given by the previous command.
+
+**Example —** `package.json`
+
+```json
+{
+  // ...
+  "dependencies": {
+    "strapi": "$version",
+    "strapi-admin": "$version",
+    "strapi-connector-bookshelf": "$version",
+    "strapi-plugin-content-manager": "$version",
+    "strapi-plugin-content-type-builder": "$version",
+    "strapi-plugin-email": "$version",
+    "strapi-plugin-graphql": "$version",
+    "strapi-plugin-upload": "$version",
+    "strapi-plugin-users-permissions": "$version",
+    "strapi-utils": "$version"
+  }
+}
+```
+
+:::tip NOTE
+Make sure to replace `$version` with the highest version given by the previous command.
+:::
+
+Finally, update your `dependencies` with one of the following commands:
+
+```bash
+yarn install
+# or
+npm install
+```
+
+## 2. Define the admin JWT Token
+
+This version comes with a new feature: Role & Permissions for the administrators. In the process, the authentication system for administrators has been updated and the `secret` used to encode the jwt token is not automatically generated anymore.
+In order to make the login work again you need to define the `secret` you want to use in `server.js`.
+
+**Example —** `config/server.js`
+
+```js
+module.exports = ({ env }) => ({
+  // ...
+  admin: {
+    auth: {
+      secret: env('ADMIN_JWT_SECRET', 'example-secret'),
```

Comment:
I am wondering if we should rather remove the default value and explain how to add the env var instead.